src/main/java/Kiwi.java

```diff
@@ -0,0 +1,180 @@
+                        handleDeadlineCommand(fullCommand);
+                        break;
+                    case EVENT:
+                        handleEventCommand(fullCommand);
+                        break;
+                    case ON:
+                        handleOnCommand(fullCommand);
+                        break;
+                    case UNKNOWN:
+                    default:
+                        throw new UnknownCommandException();
+                }
+
+            } catch (KiwiException e) {
+                ui.showError(e.getMessage());
+            } catch (Exception e) {
+                ui.showError("Something unexpected happened: " + e.getMessage());
+            }
+        }
+
+        ui.close();
+    }
+
+    private void handleByeCommand() {
+        ui.showGoodbye();
+    }
+
+    private void handleListCommand() {
+        ui.showTaskList(tasks);
+    }
+
+    private void handleMarkCommand(String command) throws KiwiException {
+        int taskNumber = Parser.parseTaskNumber(command, "mark");
+        int index = taskNumber - 1;
+
+        if (!tasks.isValidIndex(index)) {
+            throw new KiwiException("Task number " + taskNumber + " doesn't exist");
+        }
+
+        tasks.markTask(index);
+        saveTasksToStorage();
+        ui.showTaskMarked(tasks.getTask(index));
+    }
+
+    private void handleUnmarkCommand(String command) throws KiwiException {
+        int taskNumber = Parser.parseTaskNumber(command, "unmark");
+        int index = taskNumber - 1;
+
+        if (!tasks.isValidIndex(index)) {
+            throw new KiwiException("Task number " + taskNumber + " doesn't exist");
+        }
+
+        tasks.unmarkTask(index);
+        saveTasksToStorage();
+        ui.showTaskUnmarked(tasks.getTask(index));
+    }
+
+    private void handleDeleteCommand(String command) throws KiwiException {
+        int taskNumber = Parser.parseTaskNumber(command, "delete");
+        int index = taskNumber - 1;
+
+        if (!tasks.isValidIndex(index)) {
+            throw new KiwiException("Task number " + taskNumber + " doesn't exist");
+        }
+
+        Task deletedTask = tasks.deleteTask(index);
+        saveTasksToStorage();
+        ui.showTaskDeleted(deletedTask, tasks.size());
+    }
+
+    private void handleTodoCommand(String command) throws KiwiException {
+        String description = Parser.parseTodoCommand(command);
+        Todo todo = new Todo(description);
+        tasks.addTask(todo);
+        saveTasksToStorage();
+        ui.showTaskAdded(todo, tasks.size());
+    }
+
+    private void handleDeadlineCommand(String command) throws KiwiException {
+        String[] parts = Parser.parseDeadlineCommand(command);
+        String description = parts[0];
+        String deadline = parts[1];
+
+        Deadline task = new Deadline(description, deadline);
+        tasks.addTask(task);
+        saveTasksToStorage();
+        ui.showTaskAdded(task, tasks.size());
+    }
```

Comment:
```suggestion
    private void handleDeadlineCommand(String command) throws KiwiException {
        String[] args = Parser.parseDeadlineCommand(command);
        String description = args[0];
        String deadline = args[1];

        Deadline task = new Deadline(description, deadline);
        tasks.addTask(task);
        saveTasksToStorage();
        ui.showTaskAdded(task, tasks.size());
    }
```

Maybe you would like to use ```args``` instead of ```parts``` as a variable name to make clear the fact that these are the arguments of the specific command? Same with ```handleEventCommand```